src/plugins/event-plugins/ResourcePlugin.ts

```diff
@@ -134,10 +133,19 @@ export class ResourcePlugin implements Plugin {
     };
 
     recordResourceEvent = (entryData: PerformanceResourceTiming): void => {
-        if (
-            this.recordEvent &&
-            getHost(entryData.name) !== getHost(this.context.config.endpoint)
-        ) {
+        try {
+            // Ignore monitoring beacons.
+            if (
+                new URL(entryData.name).host ===
+                this.context.config.endpointUrl.host
+            ) {
+                return;
+            }
```

Comment:
I don't see how this would ever throw an error (and need a try/catch block)

Reply:
Agree -- fixed